### PR TITLE
Remove XMLSerializer export

### PR DESCRIPTION
**Pull request by domenic**
This changes the package to stop exporting the XMLSerializer class, since it was fairly useless anyway. Instead, the main module default exports the serialize() function directly, which is more straightforward and just as good for standalone consumers. jsdom can wrap serialize() in its own XMLSerializer implementation.

The serialize() function was changed from taking a boolean second parameter to taking an options object. Additionally, it no longer catches exceptions and re-throws them as DOMExceptions using the domexception package; instead it throws Errors (for well-formedness) or TypeErrors (for misuse) directly.

We also consolidate all the tests into a single file, as the division between them was fuzzy.

Finally, the package now requires Node.js v10 or later.

---

This closes #20 by superseding it.

### CODE REVIEW

**Review by ExE-Boss — APPROVED** [review]
I’d&nbsp;prefer if&nbsp;we’d&nbsp;keep the&nbsp;minimum supported&nbsp;version at&nbsp;**ES2015**, since&nbsp;we're&nbsp;not&nbsp;changing&nbsp;much and&nbsp;we’re&nbsp;not directly&nbsp;depending on&nbsp;any&nbsp;new&nbsp;features, but&nbsp;otherwise&nbsp;LGTM.

Also, bumping&nbsp;the&nbsp;minimum&nbsp;Node version&nbsp;to&nbsp;10 breaks&nbsp;**JSDOM**&nbsp;tests.

**Comment by domenic** [comment]
I tried to pick something that roughly corresponded to Node v10, but perhaps it's best to just leave it lower until we actually need or use anything from higher ES versions.

File: package.json [hunk 1]
```diff
@@ -10,25 +10,23 @@
...
-  "main": "lib/index.js",
+  "main": "lib/serialize.js",
   "scripts": {
-    "prepare": "node scripts/convert-idl.js",
-    "pretest": "node scripts/convert-idl.js",
     "test": "jest",
     "lint": "eslint ."
+  },
+  "engines": {
+    "node": ">=10"
```
Comment (ExE-Boss) [inline package.json:30]: You&nbsp;also&nbsp;need to&nbsp;remove this&nbsp;line to&nbsp;prevent the&nbsp;**JSDOM** tests&nbsp;from&nbsp;failing.

```suggestion
```

Comment (domenic) [reply package.json:30]: Not after a rebase.

**Review by pmdartus — APPROVED** [review]

**Pull request merged**